src/JobDescription.js

```diff
@@ -1,55 +1,65 @@
 import React, { Component } from 'react';
 import Container from 'react-bootstrap/lib/Container';
 import withUserValidation from './withUserValidation';
-import { createBrowserHistory } from 'history';
 import jobUtils from './jobUtils';
 import './JobDescription.css';
 
+import auth from './auth';
+
 class JobDescription extends Component {
+  state = {
+    job: null
+  };
+
+  componentDidMount() {
+    fetch(`/job/${this.props.match.params.jobId}`, { headers: auth.header() })
+      .then(res => res.json())
+      .then(data => this.setState({ job: data }));
+  }
 
   render() {
-    const history = createBrowserHistory();
-    const job = history.location.state.job;
     return (
       <Container>
```

Comment:
Maybe we just have a guard and return an error message if there is no `job`?

Reply:
did you mean like an alert or like a default element to render?

Reply:
Default element / plain text to render. Or redirect to 404 page (maybe later).